src/use-swr.ts

```diff
@@ -167,7 +167,9 @@ function useSWR<Data = any, Error = any>(
     fn = config.fetcher
   }
 
-  const initialData = cache.get(key) || config.initialData
+  if (config.initialData) cache.set(key, config.initialData, false)
```

Comment:
What happens when the component is updated and you still have initialData in the config?

Reply:
Maybe this should do something like:
```ts
let initialData = cache.get(key) ?? config.initialData
if (!cache.has(key)) cache.set(key, config.initialData, false)
```
If the key is already in cache ready it, if not use the config.initialData and update the cache.

Reply:
Sorry for the delay, took me some time to figure out how to test this in my own project.  Seems to be working fine as is without these proposed changes.

Can you clarify what you mean by "when the hook is updated".  I don't fully understand the comment.

Reply:
@sergiodxa I think I understand your point now.

If I initialize the hook with key `foo` in one part of my application with initial data `['1', '2']`, but in another part of the application init the hook, again with key `foo`, but with initial data `['3', '4']` what is the desired behavior?

@sergiodxa @pacocoursey should the first or last hook take precedence?

Reply:
Yeh if there is initialData defined you are first overwriting the cache and then reading from it. I think if already have some cached value the expected result is to get that value and then wait for a revalidation, so the second usage of the same key should receive the cached value even if that cached value is the initial value of another call of SWR.

Reply:
I agree, which means the first hook should take precedence.

Hook called with initial data: `a` → `a` cached
Hook called somewhere else with different initial data → return `a`

This also means that:

Hook called without initial data but returns data: `b` → `b` cached
Hook called somewhere else with initial data → return `b` (ignore initialdata)